libs/dyn/merge/elements_by_key.go

```diff
@@ -7,7 +7,7 @@ type elementsByKey struct {
 	keyFunc func(dyn.Value) string
 }
 
-func (e elementsByKey) Map(_ dyn.Path, v dyn.Value) (dyn.Value, error) {
+func (e elementsByKey) doMap(_ dyn.Path, v dyn.Value, mergeFunc func(a, b dyn.Value) (dyn.Value, error)) (dyn.Value, error) {
```

Comment:
Not sure what's going on here and there is no explanation on the PR. 

Perhaps this deserves a separate PR with its own explanation that the main PR can be rebased upon?

Reply:
It's a small refactoring to allow configuration merging with override (MapWithOverride) instead of merging. This is needed for Apps for now only. Indeed could make sense to refactor out in separate PR but since this one is ready to merge anyway and pretty small I'd rather keep it